### PR TITLE
Added support for user stages in stage execute command

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Added support for user stages in stage execute command

### CODE REVIEW

File: src/snowflake/cli/plugins/stage/manager.py [hunk 1]
```diff
@@ -39,6 +39,7 @@
 
 UNQUOTED_FILE_URI_REGEX = r"[\w/*?\-.=&{}$#[\]\"\\!@%^+:]+"
 EXECUTE_SUPPORTED_FILES_FORMATS = {".sql"}
+USER_STAGE = "@~"
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/plugins/stage/manager.py:42]: Nit
```suggestion
USER_STAGE_PREFIX = "@~"
```

Comment (sfc-gh-astus) [reply src/snowflake/cli/plugins/stage/manager.py]: Changed

File: src/snowflake/cli/plugins/stage/manager.py [hunk 2]
```diff
@@ -257,11 +262,14 @@ def _split_stage_path(self, stage_path: str) -> StagePathParts:
             stage_name -> stage
             directory -> dir
         """
+        directory = "/".join(Path(stage_path).parts[1:])
+
+        if self._is_user_stage(stage_path):
+            return StagePathParts(stage="@~", stage_name="@~", directory=directory)
```
Comment (sfc-gh-mraba) [inline src/snowflake/cli/plugins/stage/manager.py:268]: I wonder if there is a possibility to create a`Stage` object that would represent stage and has all the logic inside.
This way we could reference in code only to `Stage` object or its properties.

Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/plugins/stage/manager.py]: That would be useful but probably out of scope for this Pr

File: tests/stage/test_stage.py [hunk 3]
```diff
@@ -737,6 +737,46 @@ def test_execute(
...
+    [
+        ("@~", ["@~/s1.sql", "@~/a/s3.sql", "@~/a/b/s4.sql"]),
+        ("@~/s1.sql", ["@~/s1.sql"]),
+        ("@~/a", ["@~/a/s3.sql", "@~/a/b/s4.sql"]),
+        ("@~/a/s3.sql", ["@~/a/s3.sql"]),
+        ("@~/a/b", ["@~/a/b/s4.sql"]),
+    ],
+)
+@mock.patch(f"{STAGE_MANAGER}._execute_query")
+def test_execute_from_user_stage(
```
Comment (sfc-gh-turbaszek) [inline tests/stage/test_stage.py:751]: Let's add also an integration tests for this type of stage

Comment (sfc-gh-astus) [reply tests/stage/test_stage.py:752]: Added

**Pull request merged**